saleor/graphql/order/mutations/order_note_update.py

```diff
@@ -0,0 +1,44 @@
+import graphene
+
+from ....core.tracing import traced_atomic_transaction
+from ....order import OrderEvents
+from ....permission.enums import OrderPermissions
+from ...core import ResolveInfo
+from ...core.doc_category import DOC_CATEGORY_ORDERS
+from ...core.types import OrderError
+from ...plugins.dataloaders import get_plugin_manager_promise
+from ..types import OrderEvent
+from .order_note_common import OrderNoteCommon
+from .utils import get_webhook_handler_by_order_status
+
+
+class OrderNoteUpdate(OrderNoteCommon):
+    class Arguments(OrderNoteCommon.Arguments):
+        id = graphene.ID(
+            required=True,
+            description="ID of the note.",
+            name="note",
+        )
+
+    class Meta:
+        description = "Updates note of an order."
+        doc_category = DOC_CATEGORY_ORDERS
+        permissions = (OrderPermissions.MANAGE_ORDERS,)
+        error_type_class = OrderError
+        error_type_field = "order_errors"
+
+    @classmethod
+    def perform_mutation(  # type: ignore[override]
+        cls, _root, info: ResolveInfo, /, *, id: str, input
+    ):
+        qs = OrderEvent.get_model().objects.filter(type=OrderEvents.NOTE_ADDED)
+        order_event = cls.get_node_or_error(info, id, only_type=OrderEvent, qs=qs)
+        order = order_event.order
+        cleaned_input = cls.clean_input(info, order, input)
+        manager = get_plugin_manager_promise(info.context).get()
+        with traced_atomic_transaction():
+            order_event.parameters["message"] = cleaned_input["message"]
+            order_event.save()
```

Comment:
Same here. We should rather create a new event instead of updating the existing one.

Reply:
I think it's OK to update the existing event; if we want to track the history of changes, it should be displayed as one event and not flood the timeline with a new event for every change. @patrys @timuric What do you think?

Reply:
It's not ok to delete historical records. We can mark them in the UI as replaced and merge consecutive edits if they happen within a short period (say within 5 minutes if no event happened in between).

Reply:
We want to be able to remove notes that were e.g. meant to be temporary, from an order created by staff. Please check the https://github.com/saleor/saleor/issues/12397 issue for more information. I understand that historical data shouldn't be altered, but given that notes are based on `OrderEvent` it might be inconvenient to display them properly on the frontend. There would have to be some filtering/aggregating performed, although it's definitely doable.

If we choose the new event types approach, we would need to somehow store a reference to the original note that was updated/removed - is the `parameters` field in `OrderEvent` the right place for that? Perhaps the key could be called `original_event_id`? It would also have to be added to the `OrderEvent` GraphQL type.

Reply:
I think it's a UI problem to only show prominently the most recent note, not a core problem to remove the previous events from the timeline.

Reply:
Sure, I'll change the implementation to add new events instead of removing the old ones.